docs/memory.md

```diff
@@ -0,0 +1,104 @@
+performance unless your application is run in profile mode.
+
+## Memory Usage Overview Chart
+
+Is a timeseries graph to visualize the state of the Flutter memory at successive intervals of time.  Each data point on
+the chart corresponds to the timestamp (x-axis) of measured quantities (y-axis) of the heap e.g.,  usage, capacity, external,
+garbage collection and Resident Set Size.
+
+<img src="images/memory_anatomy.png" width="800" />
+
+- Legend - All collected measurements regarding the memory.  Clicking on a legend name will hide or show that data.
+- Range Selector - All memory data collected (timeseries) from the beginning of time, launch of the application.
+- Range Selector View - Detailed view of the data collected for this timeseries range (non-gray area).
+- X-axis Timestamps - Time of the collected memory information (capacity, used, external, RSS, and GC)
+- Hover Information of Collected Data - At a particular time (x-axis) the detailed collected memory data.
+- [Garbage Collection](#glossary-of-vm-terms) Occurred - Compaction of the heap occurred.
+- Event Timeline - When a user action occurred (e.g., "Snapshot" or "Reset" button clicked) see upper right buttons.
+- Snapshot - Display a list of current active memory objects (see Snapshot Classes).
+- Reset Accumulators - Reset values, to zero, under the column “Accumulator” in the Snapshot Classes list.
+- Filtering Classes - _TODO_
+- Snapshot Classes - Clicking on the Snapshot button, top right area, will display the list of current memory objects.  The memory objedts can be sorted by class name, size, allocated instances, etc.
+- Accumulator Counts Since Reset - Clicking on the Reset button, top right area, will reset the accumulated instances count.  Clicking on Snapshot after a reset will display the number of new instances allocated since last reset.  This is useful to finding memory leaks.
+- Instances of selected classes - Clicking on a class in the Snapshot Classes list will display the number of active instances.
+- Inspecting contents of an instance - _TODO_
+- Total Active Objects and Classes in the Heap - Total Classes allocated in the Heap and Total Objects (instances) in the Heap.
+
+## Memory Chart
+
+<img src="images/memory_basic_chart.png" width="800" />
+
+The chart's x-axis is a timeline of timestamps events (time series) from the VM as memory changes or the polled state of the memory
+every 500 ms.  This helps to give a live appearance on the state of the memory as the application is running.  The quantities plotted on the
+y-axis are (from top to bottom):
+- Capacity - Current capacity of the heap.
+- GC - GC has occurred.
+- Used - Objects (Dart objects) in the the heap.
+- External - memory not in the Dart heap, but is retained (memory read from a file) or decoded image in Flutter.
+
+<img src="images/memory_rss_chart.png" width="800" />
+
+To view RSS click on the RSS name in the legend.
```

Comment:
Define what RSS is a little bit in the title.
To view the Resident Set Size (RSS), click on the RSS entry in the legend.

Reply:
Is this enough?

To view RSS (Resident Set Size) click on the name RSS located in the legend.

- The Resident Set Size is used to show how much memory is allocated to a process in RAM. It does not include memory that is swapped out. It includes memory from shared libraries as long as the pages from those libraries are actually in memory. It includes all stack and heap memory.